sdk/core/Azure.Core/tests/TestFramework/RecordSession.cs

```diff
@@ -84,5 +85,36 @@ public void Sanitize(RecordedTestSanitizer sanitizer)
                 }
             }
         }
+
+        public bool IsEquivalent(RecordSession session, RecordMatcher matcher)
+        {
+            if (session == null)
+            {
+                return false;
+            }
+
+            return session.Variables.SequenceEqual(Variables) &&
+                   session.Entries.SequenceEqual(Entries, new EntryEquivalentComparer(matcher));
+        }
+
+        private class EntryEquivalentComparer : IEqualityComparer<RecordEntry>
+        {
+            private readonly RecordMatcher _matcher;
+
+            public EntryEquivalentComparer(RecordMatcher matcher)
+            {
+                _matcher = matcher;
+            }
+
+            public bool Equals(RecordEntry x, RecordEntry y)
+            {
+                return _matcher.IsEquivalentResponse(x, y);
```

Comment:
Nit - Would it be simpler to just implement this on `RecordMatcher`?

Reply:
RecordMatcher does multiple kinds of matching that involve `RecordEntry` (matches requests). So I would rather keep things more explicit.